dia/src/main.rs

```diff
@@ -0,0 +1,38 @@
+mod bindings {
+    windows::core::include_bindings!();
+}
+
+use bindings::Microsoft::Dia::*;
+use bindings::Windows::Win32::Foundation::PWSTR;
+use bindings::Windows::Win32::System::Com::*;
+use std::os::windows::prelude::OsStrExt;
+
+fn main() -> windows::core::Result<()> {
+    unsafe {
+        CoInitializeEx(std::ptr::null_mut(), COINIT_MULTITHREADED)?;
+        let dia_source: IDiaDataSource = CoCreateInstance(&DiaSource, None, CLSCTX_INPROC_SERVER)?;
+
+        let mut vec = std::env::current_exe()
+            .unwrap()
+            .as_os_str()
+            .encode_wide()
+            .collect::<Vec<u16>>();
+        vec.push(0x00);
+        vec.push(0x00);
+        dia_source.loadDataForExe(PWSTR(vec.as_mut_ptr()), None, None)?;
```

Comment:
This should accept the result of `as_os_str()` directly if you have the `alloc` and `std` features.